pyaco-core/src/classes_parser.rs

```diff
@@ -100,21 +122,32 @@ impl<'i> AtRuleParser<'i> for ClassesParser {
                     if let Ok(Token::Ident(ident)) = input.next() {
                         ret.insert(ident.to_string());
                     } else {
-                        return Err(input.new_error(BasicParseErrorKind::QualifiedRuleInvalid));
+                        return Err(input.new_error(BasicParseErrorKind::AtRuleBodyInvalid));
                     }
                 }
                 // Match any other token and ignore
                 Ok(_) => continue,
-                // Match end of input, break and return found classes
+                // Match end of input, break and return found classes if any
                 Err(BasicParseError {
                     kind: BasicParseErrorKind::EndOfInput,
                     ..
                 }) => break,
-                // Match any other error, break now and return an error
-                Err(_) => return Err(input.new_error(BasicParseErrorKind::QualifiedRuleInvalid)),
+                // Match any other error and return it
+                Err(error) => return Err(error.into()),
             }
         }
 
-        Ok(ret)
+        Ok(Some(ret))
+    }
+
+    // Simply ignores @rules without blocks, the implementation of this function
+    // is required by the `cssparser` crate if a `AtRuleType::WithoutBlock` value
+    // is returned at runtime by the `parse_prelude` function.
+    fn rule_without_block(
```

Comment:
It seems this function makes error, because it need to be override if `AtRuleParser::parse_prelude` ever returns `AtRuleType::WithoutBlock`

Reply:
Yes, that's why I added it 👍 

Notice that this restriction (that was rather weird in the first place) got lifted in cssparser 0.29